drivers/crypto/crypto_nxp_hse.c

```diff
@@ -0,0 +1,664 @@
+	session->crypto_serv_desc.hseSrv.symCipherReq.cipherAlgo = HSE_CIPHER_ALGO_AES;
+	session->crypto_serv_desc.hseSrv.symCipherReq.keyHandle = session->key_handle;
+	session->crypto_serv_desc.hseSrv.symCipherReq.sgtOption = HSE_SGT_OPTION_NONE;
+
+	ctx->drv_sessn_state = session;
+	ctx->device = dev;
+
+	return 0;
+}
+
+static int crypto_nxp_hse_cipher_free_session(const struct device *dev, struct cipher_ctx *ctx)
+{
+	struct crypto_nxp_hse_session *session;
+
+	ARG_UNUSED(dev);
+
+	session = ctx->drv_sessn_state;
+	k_mutex_lock(&session->crypto_lock, K_FOREVER);
+	memset(&session->req_type, 0, sizeof(Hse_Ip_ReqType));
+	memset(&session->crypto_serv_desc, 0, sizeof(hseSrvDescriptor_t));
+	free_session(session);
+	k_mutex_unlock(&session->crypto_lock);
+
+	return 0;
+}
+
+static int crypto_nxp_hse_sha(struct hash_ctx *ctx, struct hash_pkt *pkt, bool finish)
+{
+	struct crypto_nxp_hse_session *session = ctx->drv_sessn_state;
+	struct crypto_nxp_hse_data *data = ctx->device->data;
+	hseHashSrv_t *hash_serv = &(session->crypto_serv_desc.hseSrv.hashReq);
+
+	if (!finish) {
+		return 0;
+	}
+
+	/* Update crypto crypto description input */
+	hash_serv->pInput = HSE_PTR_TO_HOST_ADDR(pkt->in_buf);
+	hash_serv->inputLength = pkt->in_len;
+
+	k_mutex_lock(&session->crypto_lock, K_FOREVER);
+
+	if (Hse_Ip_ServiceRequest(
+		    data->mu_instance, session->channel, (Hse_Ip_ReqType *)&session->req_type,
+		    (hseSrvDescriptor_t *)&session->crypto_serv_desc) != HSE_SRV_RSP_OK) {
+		k_mutex_unlock(&session->crypto_lock);
+		return -EIO;
+	}
+
+	memcpy(pkt->out_buf, session->out_buff,
+	       *(uint32_t *)(session->crypto_serv_desc.hseSrv.hashReq.pHashLength));
+
+	k_mutex_unlock(&session->crypto_lock);
+
+	return 0;
+}
+
+static int crypto_nxp_hse_hash_begin_session(const struct device *dev, struct hash_ctx *ctx,
+					     enum hash_algo algo)
+{
+	struct crypto_nxp_hse_session *session;
+	static uint32_t out_len;
+	hseHashSrv_t *hash_serv;
+
+	if (ctx->flags & ~(CRYPTO_NXP_HSE_HASH_CAPS)) {
+		return -ENOTSUP;
+	}
+
+	session = crypto_nxp_hse_get_session(dev);
+	if (session == NULL) {
+		return -ENOSPC;
+	}
+
+	session->req_type.eReqType = HSE_IP_REQTYPE_SYNC;
+	session->req_type.u32Timeout = CRYPTO_NXP_HSE_TIMEOUT;
+
+	hash_serv = &(session->crypto_serv_desc.hseSrv.hashReq);
+
+	switch (algo) {
+	case CRYPTO_HASH_ALGO_SHA224: {
+		hash_serv->hashAlgo = HSE_HASH_ALGO_SHA2_224;
+		out_len = CRYPTO_NXP_HSE_LEN_BITS_TO_BYTES(224);
+		break;
+	}
+	case CRYPTO_HASH_ALGO_SHA256: {
+		hash_serv->hashAlgo = HSE_HASH_ALGO_SHA2_256;
+		out_len = CRYPTO_NXP_HSE_LEN_BITS_TO_BYTES(256);
+		break;
+	}
+	case CRYPTO_HASH_ALGO_SHA384: {
+		hash_serv->hashAlgo = HSE_HASH_ALGO_SHA2_384;
+		out_len = CRYPTO_NXP_HSE_LEN_BITS_TO_BYTES(384);
+		break;
+	}
+	case CRYPTO_HASH_ALGO_SHA512: {
+		hash_serv->hashAlgo = HSE_HASH_ALGO_SHA2_512;
+		out_len = CRYPTO_NXP_HSE_LEN_BITS_TO_BYTES(512);
+		break;
+	}
+	default: {
+		return -ENOTSUP;
+	}
```

Comment:
pls fix indentation

Reply:
not the wrong indentation, it is the correct format.